esmerald/transformers/signature.py

```diff
@@ -516,12 +521,19 @@ def _find_encoder(self, annotation: Any) -> Any:
             Any: The encoder found, or None if no encoder matches.
         """
         origin = get_origin(annotation)
-        for encoder in ENCODER_TYPES:
-            if not origin and encoder.is_type(annotation):
+        for encoder in ENCODER_TYPES_CTX.get():
+            if (
+                not origin
+                and hasattr(encoder, "is_type_structure")
+                and encoder.is_type_structure(annotation)
```

Comment:
The reason why all the tests are failing its because you removed the custom encoders of Esmerald. They differ a bit from Lilya, intentionally. 

Roll everything back to the way it was before. There is a reason why its using the `is_type` and not the structure. It was ALL working before.

This was not needed whatsoever

Reply:
yep, this will work, despite a bit hacky (subclass check in is_type is a bit unexpected), I think I will undo most

Reply:
Rollback everything. Esmerald is not Lilya and it operates differently. Not a hack